### ISSUE TITLE
Muse: Designing a studio for ideas

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#9136]
Muse: Designing a studio for ideas<br>
<br>
https://ift.tt/1yeDVxa<br>
<br>
<br>
<br>
<div>
<div>
<div>
<div>
<header>
<h1>Muse</h1>
<h2>Designing a studio for ideas</h2>
<p>Muse is a prototype iPad app for reading, thinking, and developing your ideas. Starting with inspiration from physical workspaces, it tries to be a repository for your creative inputs; helps you see and connect all your raw material to make new creative leaps; and enables active reading of PDFs and other media via annotations and excerpting. All of this is in a fast, fluid user interface to create a sense of empowerment and excitement for you as a creative person.</p>
<figure><video alt="Some capabilities of the Muse studio for ideas iPad application" src="https://www.inkandswitch.com/muse/static/muse-tour.mp4#t=0.01"></video>
<figcaption>
<h4>A brief tour</h4>
<p>Some capabilities of the Muse studio for ideas iPad application</p>
</figcaption>
</figure>
<p>Muse was created at the <a href="https://www.inkandswitch.com">Ink &amp; Switch research lab</a>, continuing our work on tablet-based tools for power users. This article gathers our findings on the design side of this project; a forthcoming companion article will cover the technical challenges of implementing a speed-of-thought zooming UI at 120&nbsp;Hz on iOS.</p>
<p>We welcome your feedback: <a title="Tweet us @inkandswitch" href="https://twitter.com/inkandswitch">@inkandswitch</a> or <span><a title="Send us an Email" href="https://www.inkandswitch.com/cdn-cgi/l/email-protection#81e9e4ededeec1e8efeae0efe5f2f6e8f5e2e9afe2eeec"><span>[email&nbsp;protected]</span></a>.</span></p>
</header>
<article>
<p>Creative professionals need a place to develop their ideas. Digital authoring tools like Google Docs or Photoshop are designed to express ideas you&rsquo;ve already developed, and productivity tools like email, calendar, or todo lists are good for tracking and administration. Surprisingly few digital tools exist for the early, freeform phase of ideation.</p>
<p>Building on the foundation of <a href="https://www.inkandswitch.com/capstone-manuscript.html">our earlier user research</a>, we continue to explore tablet interfaces as a freeform thinking space. For this iteration, we took inspiration from physical spaces for idea gestation and crafting: libraries, workshops, drafting tables, and artists&rsquo; studios.</p>
<p>Creative people tend to nest. A professor in their classroom, a writer in their home office, a woodworker in their shop, an artist in their studio&mdash;these spaces are full of work surfaces like desks and drafting tables, drawers and pegboards full of tools, pinboards and chalkboards and whiteboards, scraps of paper, photos, books, printouts, works-in-progress, post-its, and more. They are messy, informal, mixed together, freeform, and personal.</p>
<figure><img alt="A workspace that&amp;rsquo;s personal and informal. (source)" src="https://www.inkandswitch.com/muse/static/workspace.jpg">
<figcaption>
<h4>Physical workspace</h4>
<p>A workspace that&rsquo;s personal and informal. (<a href="https://huckberry.com/journal/posts/creators-project-casey-neistat">source</a>)</p>
</figcaption>
</figure>
<p>Compare these to our digital workspaces: what we see on the screens of our computers, tablets, and phones. Lists and grids of items, neatly arranged by the computer (perhaps sorted by date or name). Media types are often siloed (text notes in one place, web pages in another, PDFs and their highlights in still another) and difficult to view together. We don&rsquo;t entirely choose what goes into our workspaces and things don&rsquo;t tend to stay where we left them. In short, digital workspaces are structured, sterile, and impersonal.</p>
<p><a href="https://fs.blog/2014/08/steve-jobs-on-creativity/">Creativity is about making connections.</a> This seems to demand a freeform, fluid space where creative fodder can be mixed together and sorted to the user&rsquo;s liking. So why are freeform environments so rare in digital workspaces?</p>
<figure>
<figcaption>
<p>Physical and digital workspaces typically have different properties. We explore bringing physical notions to a digital tool.</p>
</figcaption>
<table>
<tbody>
<tr>
<th>
<p>Physical workspace</p>
</th>
<th>
<p>Digital workspace</p>
</th>
</tr>
<tr>
<td>
<p>Arranged by you</p>
</td>
<td>
<p>Arranged by the computer</p>
</td>
</tr>
<tr>
<td>
<p>Freeform, messy, informal, playful</p>
</td>
<td>
<p>Neat, sorted, structured, sterile</p>
</td>
</tr>
<tr>
<td>
<p>Write notes, draw sketches, underline, highlight anywhere on the page</p>
</td>
<td>
<p>Only make marks in spaces delineated as drawing canvases by the computer</p>
</td>
</tr>
<tr>
<td>
<p>Rip out pages, photocopy article excerpts, tape sketches and printouts to the wall, put things side-by-side</p>
</td>
<td>
<p>One document / page at a time (mobile)</p>
<p>Flip between document tabbed views</p>
<p>Arrange windows side-by-side (desktop)</p>
</td>
</tr>
<tr>
<td>
<p>Everything there was brought by you</p>
<p>Things stay where you left them</p>
</td>
<td>
<p>Computer refreshes web pages, reorders document lists, or suggests new content</p>
<p>Window or tab arrangement often lost between sessions</p>
</td>
</tr>
<tr>
<td>
<p>Everything together</p>
</td>
<td>
<p>Photos in one app, PDFs in another, e-books in a third, text files elsewhere still</p>
</td>
</tr>
</tbody>
</table>
</figure>
<p>This led to our concept for this project. Inspired by physical workspaces, we set out to build a studio for ideas.</p>
<p>Based on the above comparison, we set out with the following design goals:</p>
<ul>
<li>
<strong>A repository for creative inputs.</strong> The studio is a place to collect raw material as input to your thinking. This means everything together, with no media silos. If you get a critical mass of documents into the studio, connections will naturally form. These connections produce new ideas that can be captured in the studio&mdash;a virtuous cycle producing yet more fodder for generation of future ideas.</li>
<li>
<strong>Freeform and arranged by you.</strong> A spatial environment allows you to pull out excerpts, places things side by side, arrange them in a loose, freeform, fluid way.</li>
<li>
<strong>Multimedia.</strong> Text, photos, sketches, PDFs, web pages, diagrams&mdash;all together to reflect your variety in creative inputs and your own thinking.</li>
<li>
<strong>Inking everywhere.</strong> Drawings, annotations, <a href="https://www.newyorker.com/books/page-turner/the-marginal-obsession-with-marginalia">marginalia</a>. Not only can you see every media type together, but you can ink on everything without restriction. This supports everything from playful doodling to the serious study of <a href="https://www.sciencemag.org/careers/2016/03/how-seriously-read-scientific-paper">active reading</a>.</li>
<li>
<strong>Excerpting.</strong> In addition to freeform arrangement of entire documents, you may want to pull out just a sentence, a paragraph, a page, or a diagram from a longer document.</li>
</ul>
<p>Added to these are some of Ink &amp; Switch&rsquo;s general design principles for digital tools:</p>
<ul>
<li>
<strong>No chrome.</strong> Avoid toolbars, buttons, or other administrative debris. Just you and your work. The studio presents only a blank page, quietly inviting you to capture your thoughts and develop your ideas.</li>
<li>
<strong>Fast.</strong> The studio should support creators moving at their natural speed of thought. It should never break the connection between user inputs and the reactions they observe from the tool. As an industry, we&rsquo;ve underestimated how harmful <a href="https://www.inkandswitch.com/slow-software.html">slow software</a> is for the creative process. It&rsquo;s so discouraging and uncomfortable to do creative work when you&rsquo;re waiting, looking at spinners, unsure if your last action registered. The studio for ideas should have no spinners, no waiting, and run at 120 frames per second. Documents open instantly. No command requires press-and-hold or other built-in delays.</li>
<li>
<strong>Use the full command gesture space.</strong> As a <a href="https://medium.com/@hirodusk/the-ipad-as-a-fast-precise-tool-for-creativity-21384ea18659">fast, precise tool</a> for creative professionals, we should use every capability at our disposal. That means all ten fingers, and the stylus as a distinct input. The stylus is required.</li>
</ul>
<p>If the studio can achieve these goals, we hope it means you, the user, will be able to create more connections among ideas that before would have been siloed. That in turn means you can develop your ideas to the very fullest and thus reach your potential as a thinking, creating human.</p>
<p>Thus we set out to build Muse, a studio for ideas inspired by physical workspaces.</p>
<p>We have not achieved all the ideals set out in the previous section, but we did hit many of them. So now we&rsquo;ll take a tour of the prototype and describe what we learned from seeing this in use by our team internally and in external users tests.</p>
<p>A space for creativity requires raw material. This material is the fodder for our minds to <a href="https://medium.com/key-lessons-from-books/the-key-lessons-from-where-good-ideas-come-from-by-steven-johnson-1798e11becdb">generate new ideas</a>.</p>
<p>Thus our studio for ideas must make ingestion of different media types easy and painless. One ingestion route we focused on was users doing research on a web browser on their desktop computer.</p>
<p>The Muse web clipper is a Chrome extension that can capture fragments of text, images, whole web pages, or PDFs. Users right-click on web content and use &ldquo;Clip to Muse&rdquo; to clip a copy of that content to their tablet.</p>
<figure><video alt="Using the web clipper to add an image and then a text snippet to Muse. Clippings appear instantly on the user&rsquo;s scratch space (the &ldquo;shelf&rdquo;). " src="https://www.inkandswitch.com/muse/static/clipper.mp4#t=0.01"></video>
<figcaption>
<h4>Clip images and text</h4>
<p>Using the web clipper to add an image and then a text snippet to Muse.</p>
<p>Clippings appear instantly on the user&rsquo;s scratch space (the &ldquo;<a href="https://www.inkandswitch.com/muse/#shelf">shelf</a>&rdquo;).</p>
</figcaption>
</figure>
<p>We designed the clipper so that users have zero choices when clipping. There are no questions about what format to clip, where to save the content etc. The user simply says &ldquo;I want to keep this&rdquo;. Any further steps here would make it harder for users to preserve content that&rsquo;s important to them.</p>
<p>While we believe the Muse web clipper is useful, content ingestion was not the main focus of Muse and we don&rsquo;t expect that the web clipper as-is would be sufficient for real use. We do think such a web extension would be one piece of the solution for a production-grade thinking tool.</p>
<p>The heart of the studio ideas is reading, sifting, sorting, and making sense of the raw material the user has ingested.</p>
<p>We chose a spatial layout for Muse: mixed-media cards organized with a boards metaphor.</p>
<figure><img alt="A Muse board with various cards, including nested boards." src="https://www.inkandswitch.com/muse/static/board.jpg">
<figcaption>
<h4>Example board</h4>
<p>A Muse board with various cards, including nested boards.</p>
</figcaption>
</figure>
<p>There are some precedents for this type of interface. <a href="https://milanote.com">Milanote</a>, for example, uses a cards-on-boards model. What&rsquo;s unique about Muse is how we tried to design the interface to reflect and leverage all the subtle but important aspects of how people work in physical space.</p>
<p>Users can easily move, resize, create, delete, and copy cards in Muse.</p>
<figure><video alt="A low-angle grip on the stylus drags cards. More on grips in Stylus tool selection below." src="https://www.inkandswitch.com/muse/static/move-card.mp4#t=0.01"></video>
<figcaption>
<h4>Move cards</h4>
<p>A low-angle grip on the stylus drags cards. More on grips in <a href="https://www.inkandswitch.com/muse/#stylus-tool-selection">Stylus tool selection</a> below.</p>
</figcaption>
</figure>
<p>We chose to use the stylus for moving and resizing because it&rsquo;s more precise than fingers. This is a good example of where a stylus-required model gives greater flexibility in interaction design.</p>
<figure><video alt="Cards resize when dragged from the corner. Images have fixed aspect ratios and the resize operation respects this." src="https://www.inkandswitch.com/muse/static/resize-card.mp4#t=0.01"></video>
<figcaption>
<h4>Resize cards</h4>
<p>Cards resize when dragged from the corner. Images have fixed aspect ratios and the resize operation respects this.</p>
</figcaption>
</figure>
<p>In terms of creating cards, we think it&rsquo;s important for you to be able to start your creative work without ceremony. Our benchmark for ease is turning to a new page in a paper notebook. With that, no idea will be squashed by the hassle of getting started. In Muse we achieve this instant-start by dragging from the edge of the screen to create a new nested board.</p>
<figure><video alt="Pulling in new a blank board&mdash;as a card nested in the current board&mdash;from the edge of the screen. Users can create and place the card with the same hand movement." src="https://www.inkandswitch.com/muse/static/create-board.mp4#t=0.01"></video>
<figcaption>
<h4>Create card</h4>
<p>Pulling in new a blank board&mdash;as a card nested in the current board&mdash;from the edge of the screen. Users can create and place the card with the same hand movement.</p>
</figcaption>
</figure>
<p>Copying cards uses a related gesture, placing one finger on the card to be copied and pulling the copy out with the stylus.</p>
<figure><video alt="The new copy fades in as the gesture proceeds. It fades back out if the operation is canceled by lifting the stylus early." src="https://www.inkandswitch.com/muse/static/copy-card.mp4#t=0.01"></video>
<figcaption>
<h4>Copy card</h4>
<p>The new copy fades in as the gesture proceeds. It fades back out if the operation is canceled by lifting the stylus early.</p>
</figcaption>
</figure>
<p>Deletion is the inverse of creating&mdash;drag a card off the screen.</p>
<figure><video alt="Flick cards to the edge to remove them." src="https://www.inkandswitch.com/muse/static/delete-card.mp4#t=0.01"></video>
<figcaption>
<h4>Delete card</h4>
<p>Flick cards to the edge to remove them.</p>
</figcaption>
</figure>
<p>A subtle benefit of this design is that you can undo a creation in mid-flight; just drag the card off the edge.</p>
<figure><video alt="The user starts to create a new board, then changes their mind and flicks it away." src="https://www.inkandswitch.com/muse/static/create-then-throw.mp4#t=0.01"></video>
<figcaption>
<h4>Create then discard</h4>
<p>The user starts to create a new board, then changes their mind and flicks it away.</p>
</figcaption>
</figure>
<p>The gestures can even be used to play around with your content, e.g. by creating a copy and throwing it away repeatedly. This may seem trivial, but it helps make the app feel more personal and less fragile to users, which we think encourages relaxed, creative thinking:</p>
<figure><video alt="A user casually creates and discards ephemeral copies of a card." src="https://www.inkandswitch.com/muse/static/copy-then-throw.mp4#t=0.01"></video>
<figcaption>
<h4>Copy then discard</h4>
<p>A user casually creates and discards ephemeral copies of a card.</p>
</figcaption>
</figure>
<p>There are a few downsides to this gesture set. One is that the create gesture doesn&rsquo;t obviously scale to multiple document types. If another document type was available in addition to boards&mdash;say, tables&mdash;we&rsquo;d need some way to indicate which type to create. The copy gesture has some ambiguity between copy-by-value vs. copy-by-reference. The deletion gesture makes hard deletes too easy to do. We&rsquo;d probably address this last issue with a combination of undo and changing delete to &ldquo;archive&rdquo; that sent documents to a staging area, as in Gmail.</p>
<p>Muse uses <a href="https://en.wikipedia.org/wiki/Zooming_user_interface">zoom-based navigation</a> to leverage users&rsquo; innate spatial thinking, visual processing, and hand-eye coordination.</p>
<figure><video alt="The app responds quickly to any pinches by initiating a zoom in the appropriate direction. When a gesture is finished the view smoothly transitions to the nearest stable zoom level. The user can &ldquo;peek&rdquo; into documents by zooming in and then back out in one gesture. " src="https://www.inkandswitch.com/muse/static/zoom-nested-boards.mp4#t=0.01"></video>
<figcaption>
<h4>Pinch to zoom</h4>
<p>The app responds quickly to any pinches by initiating a zoom in the appropriate direction. When a gesture is finished the view smoothly transitions to the nearest stable zoom level.</p>
<p>The user can &ldquo;peek&rdquo; into documents by zooming in and then back out in one gesture.</p>
</figcaption>
</figure>
<p>Any time a user initiates, adjusts, or releases a zooming pinch, the interface responds instantly. This gives a fluid feeling and builds the user&rsquo;s confidence in the interface.</p>
<p>Relatedly, there are no discrete navigation jumps. As a consequence you can always initiate a partial movement and then gradually roll it forward and backwards until you reach some stable level. It also means you never lose your orientation, as every successive screen flows smoothly from the last (and you can always pinch in to get back).</p>
<p>One question we faced with a zooming interface was how to represent boards within boards. Just scaling down the board would make an inner board hard to distinguish as it&rsquo;d be the same color as the surrounding board. To solve this we use a slight haze over distant boards, so that they appear darker the further away they are. This provides the necessary visual distinction between outer and inner boards without obscuring the content of the inner boards. It can also be incrementally animated as the zoom is in progress.</p>
<figure><video alt="Zooming in and out on a nested board shows how a slight haze denotes nested boards, fading away as the board approaches the top zoom level." src="https://www.inkandswitch.com/muse/static/nested-board-haze.mp4#t=0.01"></video>
<figcaption>
<h4>Nesting haze</h4>
<p>Zooming in and out on a nested board shows how a slight haze denotes nested boards, fading away as the board approaches the top zoom level.</p>
</figcaption>
</figure>
<p>An important implication of the zooming design is that there&rsquo;s no &ldquo;opening&rdquo; or &ldquo;closing&rdquo; documents. You just get close to them and they&rsquo;re there, and when you&rsquo;re far enough away they&rsquo;re effectively invisible. This reflects how documents work in the real world; to read a piece of paper you just look at it. You don&rsquo;t need to load your paper or open a paper-reading app first.</p>
<p>Users can hold cards they&rsquo;re moving even as they&rsquo;re zooming. This is an example of the interface taking advantage of the user using two hands&mdash;one for touch and one for the stylus&mdash;at the same time.</p>
<figure><video alt="Once a move operation has started, the card stays with the user as they navigate. Moves can therefore carry cards across boards as well as within them." src="https://www.inkandswitch.com/muse/static/pickup-and-transport.mp4#t=0.01"></video>
<figcaption>
<h4>Pick up and transport</h4>
<p>Once a move operation has started, the card stays with the user as they navigate. Moves can therefore carry cards across boards as well as within them.</p>
</figcaption>
</figure>
<p>Zooming interfaces are not new. The <a href="http://www.cl.cam.ac.uk/techreports/UCAM-CL-TR-574.pdf">first one</a> was built by Ivan Sutherland in 1962. Zooming is familiar today in maps apps like Google Maps. Indeed we deliberately borrow the pinch and double-tap gestures from these apps. However, as far as we know, Muse&rsquo;s combination of zooming interface, continuous design, concurrent stylus and hand gestures, and very high responsiveness is novel in a productivity tool.</p>
<p>Users quickly learned the interface and reported it made intuitive sense. We&rsquo;d need more testing with full user corpuses to better assess the true potential of zooming user interfaces for document management, but we think they&rsquo;re promising.</p>
<p>In physical and paper workspaces, artisans use a variety of techniques for keeping current work close at hand: desktop spaces (a metaphor used in most workstation platforms but absent in touch platforms), drawers that slide out, post-its or other items taped or pinned nearby, inboxes.</p>
<p>In Muse, we try to replicate this concept of an informal scratch space with something we call the shelf. The shelf can be summoned and dismissed with a swipe, and it moves with you as you zoom around the space.</p>
<figure><video alt="A user summons the shelf, which moves with them as they navigate. By dragging the shelf its position is adjusted. The shelf can also be flicked to move it, via momentum, more than the distance traveled by the contacting finger." src="https://www.inkandswitch.com/muse/static/shelf.mp4#t=0.01"></video>
<figcaption>
<h4>Summoning the shelf</h4>
<p>A user summons the shelf, which moves with them as they navigate. By dragging the shelf its position is adjusted. The shelf can also be flicked to move it, via momentum, more than the distance traveled by the contacting finger.</p>
</figcaption>
</figure>
<p>One use of the shelf is receiving cards from the web clipper described above. Once an item lands on the shelf, you can move it to its desired location on your board.</p>
<figure><video alt="After these cards land on the shelf from the web clipper, the user moves them to their spots on the current board." src="https://www.inkandswitch.com/muse/static/from-shelf-to-board.mp4#t=0.01"></video>
<figcaption>
<h4>From shelf to board</h4>
<p>After these cards land on the shelf from the web clipper, the user moves them to their spots on the current board.</p>
</figcaption>
</figure>
<p>The shelf also acts as a copy-paste buffer. It differs from a traditional copy-paste buffer in that a) the contents are visible to the user, b) it can hold multiple items, and c) the user can choose to use a subset of the items in the buffer.</p>
<p>The shelf can be adjusted at the same time as a card is being moved. This allows you to e.g. scoop up a card with a moving shelf, yank the shelf out from under a card to place it on a board, or do other compound movements.</p>
<figure><video alt="A user holds a card while moving the shelf out from under it." src="https://www.inkandswitch.com/muse/static/move-card-and-shelf.mp4#t=0.01"></video>
<figcaption>
<h4>Compound gestures</h4>
<p>A user holds a card while moving the shelf out from under it.</p>
</figcaption>
</figure>
<p>The shelf supports zooming, similarly to the main view. Thus it&rsquo;s possible to make a single item take up all of the shelf space, so the shelf can enable split-screen viewing of two items at the same time. An example of when this is useful is reading a PDF in the main view and taking notes in the shelf view.</p>
<figure><video alt="Leveraging the shelf to take notes while reading a PDF. Compare to the iOS Split View." src="https://www.inkandswitch.com/muse/static/notes-pdf.mp4#t=0.01"></video>
<figcaption>
<h4>Multi-document view</h4>
<p>Leveraging the shelf to take notes while reading a PDF. Compare to the <a href="https://support.apple.com/en-us/HT207582">iOS Split View</a>.</p>
</figcaption>
</figure>
<p>Overall we think the shelf is a powerful construct, especially on mobile where such scratch space is traditionally lacking.</p>
<p>Pinboards, whiteboards, print books, post-its, and other forms of paper workspaces offer not just freeform arrangement of their contents, but also the ability to scribble anywhere. Highlighted or underlined passages, marginalia, even doodles in the margin help make an informal play space where ideas can flourish.</p>
<p>For this reason, our goal for Muse was simple: you can draw on anything.</p>
<p>On boards, inking can be used for e.g. free-form notes and drawings.</p>
<figure><video alt="Basic inking on a board. The feel is similar to sheet of paper in a notebook." src="https://www.inkandswitch.com/muse/static/drawing.mp4#t=0.01"></video>
<figcaption>
<h4>Freeform sketching</h4>
<p>Basic inking on a board. The feel is similar to sheet of paper in a notebook.</p>
</figcaption>
</figure>
<p>In addition to supporting inking-as-primary-content, we also wanted to leverage ink for annotations and metadata. On a board, for example, one could use ink to label cards.</p>
<p>In Muse it&rsquo;s possible to write on any media type, not just boards. For example on PDFs, images, text, and web clippings.</p>
<figure><video alt="Annotating a PDF." src="https://www.inkandswitch.com/muse/static/annotate-pdf.mp4#t=0.01"></video>
<figcaption>
<p>Annotating a PDF.</p>
</figcaption>
</figure>
<figure><video alt="Inking on an image annotates it." src="https://www.inkandswitch.com/muse/static/annotate-image.mp4#t=0.01"></video>
<figcaption>
<p>Inking on an image annotates it.</p>
</figcaption>
</figure>
<figure><video alt="Plain text can be annotated." src="https://www.inkandswitch.com/muse/static/annotate-text.mp4#t=0.01"></video>
<figcaption>
<p>Plain text can be annotated.</p>
</figcaption>
</figure>
<figure><video alt="It&rsquo;s even possible to annotate web clippings, which are not usually thought of as amenable to inking." src="https://www.inkandswitch.com/muse/static/annotate-web-clipping.mp4#t=0.01"></video>
<figcaption>
<p>It&rsquo;s even possible to annotate web clippings, which are not usually thought of as amenable to inking.</p>
</figcaption>
</figure>
<p>Inking is an intuitive way to engage with content on the tablet form factor. We think supporting inking across all media types brings benefits both in terms of improving processing of the content with active reading and in the more concrete sense of being able to mark up the documents.</p>
<p>You can even take notes on the shelf. We sometimes use this in the same way you might use a sticky on your desk or notes in the back cover of a book.</p>
<figure><img alt="A user wrote quick todo notes on their shelf." src="https://www.inkandswitch.com/muse/static/inking-on-shelf.jpg">
<figcaption>
<h4>Inking on shelf</h4>
<p>A user wrote quick todo notes on their shelf.</p>
</figcaption>
</figure>
<p>An obvious question in an inking interface is how the ink looks. What style lines are used, with what thickness, and what color? Can these be changed? In Muse we&rsquo;ve experimented with a highly opinionated approach where the tool supports only one ink type for any given media type, with each type carefully designed for the context it&rsquo;ll be used in. The goal was to give users a consistently great inking experience without having to introduce chrome, modes, options, etc. that take up screen space and mental energy.</p>
<figure><img alt="PDF Expert offers extensive customization of the inking experience. The cost is chrome cluttering the view, a multi-step configuration process, and modes in the interface. We wanted to explore a different set of tradeoffs in Muse." src="https://www.inkandswitch.com/muse/static/pdf-expert-inking.jpg">
<figcaption>
<h4>Complex inking interface</h4>
<p><a href="https://pdfexpert.com">PDF Expert</a> offers extensive customization of the inking experience. The cost is chrome cluttering the view, a multi-step configuration process, and modes in the interface. We wanted to explore a different set of tradeoffs in Muse.</p>
</figcaption>
</figure>
<p>Overall we feel this approach worked pretty well. For lots of uses one ink style is all you need. Not seeing or thinking about ink style buttons, width sliders, etc. was great. The thing users most wanted choice on was color. Future design work could explore how to support a limited amount of choice in ink types (say, among a few colors) without creating the interface problems of tablet apps with arbitrarily customizable inking.</p>
<p>A last design consideration for ink is how to erase it. To engage the eraser a user presses one finger on the screen while using the stylus with a typical writing grip (see <a href="https://www.inkandswitch.com/muse/#stylus-tool-selection">below</a> for a full discussion of stylus tool selection). Additionally, the &ldquo;size&rdquo; of the eraser grows with the amount of pressure applied to the stylus. This avoids another bit of UI chrome (an eraser-size-selector) and proved easy for users to adopt:</p>
<figure><video alt="Erasing ink, first with light pressure for fine adjustments and then heavy, fast strokes to quickly clear a large area." src="https://www.inkandswitch.com/muse/static/erasing.mp4#t=0.01"></video>
<figcaption>
<h4>Erasing</h4>
<p>Erasing ink, first with light pressure for fine adjustments and then heavy, fast strokes to quickly clear a large area.</p>
</figcaption>
</figure>
<p>For many of our target users, PDFs are a foundational media type. Professors review papers; architects use PDFs of floorplans; attorneys read and annotate legal documents. For this reason, support for PDFs was a key feature of Muse.</p>
<p>A useful property of physical books and papers is that you can scan them quickly. For example:</p>
<ul>
<li>Identify books at a glance, based on the color, layout, etc. of their cover.</li>
<li>Flip through the book to get a sense of the content.</li>
<li>Find a known image or annotation by scanning quickly through the pages.</li>
<li>Open up to a random page and start reading to get a sample of the book.</li>
<li>Judge how many pages there are at a glance, based on thickness.</li>
</ul>
<p>We think these &ldquo;scanning&rdquo; actions are critical for users. They allow you to do important operations quickly and in a way that leverages our powerful visual pattern recognition facilities.</p>
<p>These operations are not possible in the classic computer PDF reader experience: a filename in small text next to a generic PDF icon, which when opened leads to a PDF reader that displays one page at a time with linear scrolling.</p>
<figure><video alt="A user&rsquo;s directory of PDF papers. It&rsquo;s hard to get a sense of the papers from the outside, and opening them only shows the first page, leaving e.g. the length and typical page contents impossible to quickly gauge." src="https://www.inkandswitch.com/muse/static/pdf-folder.mp4#t=0.01"></video>
<figcaption>
<h4>Folder of PDFs</h4>
<p>A user&rsquo;s directory of PDF papers. It&rsquo;s hard to get a sense of the papers from the outside, and opening them only shows the first page, leaving e.g. the length and typical page contents impossible to quickly gauge.</p>
</figcaption>
</figure>
<p>We wanted to support these scanning operations in Muse. We accomplished this with PDF overviews.</p>
<figure><video alt="A board of PDF papers in Muse comparable to the desktop data above. Muse provides much richer context and allows users to change their level of focus quickly and continuously." src="https://www.inkandswitch.com/muse/static/scan-pdfs.mp4#t=0.01"></video>
<figcaption>
<h4>Board of PDFs</h4>
<p>A board of PDF papers in Muse comparable to the desktop data above. Muse provides much richer context and allows users to change their level of focus quickly and continuously.</p>
</figcaption>
</figure>
<p>When viewed from the outside, these overviews identify the documents and give users get a sense of what to expect. The first page of the PDF is enlarged as it usually has important information (e.g. title, cover image). Users can also get a sense of the shape of the document&mdash;for example, how much images vs. text they contain&mdash;as well as see whether it is short or longer than one overview screen.</p>
<p>Within the overview, users can quickly scroll through the document. Once they find of page of interest, they can peek in smoothly to confirm it&rsquo;s what they&rsquo;re looking for, and either continue zooming in to the single-page reader or back out to the overview.</p>
<figure><img alt="We also considered a format with a scrollable column of thumbnails, as in macOS Preview. But we find this an unsatisfactory no-man&rsquo;s land. If you&rsquo;re reading a page, you just want to be looking at the content (not a bunch of thumbnails). And if you&rsquo;re scanning for a particular location you want the benefit of all available screen real estate to scan quickly. We therefore found the two seperate views a better fit." src="https://www.inkandswitch.com/muse/static/preview.jpg">
<figcaption>
<h4>Column of thumbnails</h4>
<p>We also considered a format with a scrollable column of thumbnails, as in <a href="https://support.apple.com/guide/preview/welcome/mac">macOS Preview</a>. But we find this an unsatisfactory no-man&rsquo;s land. If you&rsquo;re reading a page, you just want to be looking at the content (not a bunch of thumbnails). And if you&rsquo;re scanning for a particular location you want the benefit of all available screen real estate to scan quickly. We therefore found the two seperate views a better fit.</p>
</figcaption>
</figure>
<p>In user tests, the PDF overview was one of the most appreciated features. Users were often surprised as the overview is not like other tools they&rsquo;ve used, but with some use they came to appreciate it.</p>
<p>One way to do active reading of a document is annotations: making notes or circling relevant bits. But another feature helpful here is excerpting (also known as highlighting, quoting, or <a href="https://en.wikipedia.org/wiki/Transclusion">transclusions</a>).</p>
<p>For example you might want to excerpt facts you&rsquo;ll be citing, graphics to send to a colleague, or references to chase down at the library. For this use case Muse provides first-class excerpts.</p>
<figure><video alt="Selections that begin on text are saved in plain-text format, while excerpts capturing an arbitrary space are saved as images." src="https://www.inkandswitch.com/muse/static/excerpting.mp4#t=0.01"></video>
<figcaption>
<h4>Image and text excerpts</h4>
<p>Selections that begin on text are saved in plain-text format, while excerpts capturing an arbitrary space are saved as images.</p>
</figcaption>
</figure>
<p>To create an excerpt, you drag across the PDF with a stylus using a low-angle grip (see <a href="https://www.inkandswitch.com/muse/#h.g9td0st1gjs5">below</a> for more on grips). By default, this creates a snapshot-style excerpt, which is visually similar to what you&rsquo;d get taking a screenshot over the same area. If you start the excerpt gesture on a piece of text, a text excerpt is created which pulls out the text content in a format-independent way.</p>
<p>Excerpts are first-class cards like boards, images, and their source PDFs. You can e.g. move them around and zoom in to them as full-screen items.</p>
<figure><video alt="Excerpt cards are manipulated like any other card in Muse." src="https://www.inkandswitch.com/muse/static/organizing-excerpts.mp4#t=0.01"></video>
<figcaption>
<h4>Organizing Excerpts</h4>
<p>Excerpt cards are manipulated like any other card in Muse.</p>
</figcaption>
</figure>
<p>Excerpts also know where they came from, and can take you back to that context.</p>
<figure><video alt="A user navigates back to the source of an excerpt by pressing a black button on the excerpt card. By default, the movement is just a peek, and the user returns to their previous location when letting go. If they scroll with their finger while on the source document, they remain there. " src="https://www.inkandswitch.com/muse/static/wormhole.mp4#t=0.01"></video>
<figcaption>
<h4>Wormholes</h4>
<p>A user navigates back to the source of an excerpt by pressing a black button on the excerpt card.</p>
<p>By default, the movement is just a peek, and the user returns to their previous location when letting go. If they scroll with their finger while on the source document, they remain there.</p>
</figcaption>
</figure>
<p>When you press the black dot on the bottom-right corner of an excerpt, you &ldquo;wormhole&rdquo; to the source PDF for that excerpt, and then to the individual page, with the excerpted area highlighted. You can then navigate around the source PDF as usual. If you pinch to go back, you&rsquo;ll go first to the PDF overview page; a second pinch goes back to the page you warped from. Thus the invariant of pinching to go back where you were before is preserved. Coming in through the PDF overview also gives you some quick context on the source document as you&rsquo;re entering.</p>
<p>Note that screenshot-style excerpts are currently just bitmaps. Thus they lose the underlying text / vector information present in their source PDF. This is an artifact of our initial implementation. One could imagine preserving the vector information in a PDF-like way in excerpts, perhaps even as a little PDF. Then users would be able to do things like copy text from a screenshot-style excerpt.</p>
<p>We designed excerpting to be orthogonal from annotation. So you can annotate a section without excerpting it, or visa versa, or you can annotate a section and then take an excerpt with that annotation. We wanted to give users control over these two axes independently. It&rsquo;s tempting to combine these operations to save a step in the case the user wants both. But users often want to just e.g. annotate documents for their own active reading purposes, and separately excerpt key pieces they need for other work or to communicate with collaborators. We therefore think it&rsquo;s appropriate for them to be orthogonal, though there may be more clever approaches here.</p>
<figure>
<figcaption>
<h4>Kindle highlights</h4>
<p><a href="https://www.amazon.com/kindle-dbs/fd/kcp">Kindle</a> combines excerpting and annotations with &ldquo;highlights&rdquo;. This is convenient at times, but prevents the user from doing only one or the other."</p>
</figcaption>
</figure>
<p>We think first-class excerpts are a promising concept. There&rsquo;s more to iterate on here, for example you should probably be able to excerpt any document type, not just PDFs. But the basic idea of pulling out excerpts and being able to manipulate them with all the powers of your tool is appealing, as is the ability to reference back to the original source.</p>
<p>Perhaps the most powerful example of excerpting in a productivity tool is <a href="https://www.liquidtext.net">LiquidText</a>, from which we draw inspiration. In Muse we&rsquo;ve tried to combine such excerpting capabilities with the the design goal of managing all content types (including source documents and excerpts from them) in the same way, with the same tools (i.e. as movable cards on boards). We think this unified approach is promising but further user testing is needed.</p>
<p>Text is an important data type for creative work. In Muse, users create text cards by clipping a text string from a web page or by creating a text excerpt from a PDF. We tried a few design approaches with the resulting text cards in Muse.</p>
<p>One was to bold the first line of text cards. This provides a natural title without having to introduce markup, formatting buttons, etc. into the interface. Cards with this styling have more structure.</p>
<figure><img alt="Many cards on the same board feel less like endless blobs of text compared to a single top-to-bottom text document." src="https://www.inkandswitch.com/muse/static/text-cards.jpg">
<figcaption>
<h4>Multiple text cards</h4>
<p>Many cards on the same board feel less like endless blobs of text compared to a single top-to-bottom text document.</p>
</figcaption>
</figure>
<p>Another choice was to always apply a reasonable line width, regardless of the dimensions of the cards or level of zoom. This prevents unreadable text layouts, like lines that span the whole width of an iPad screen. Another effect of the stable layout is making it possible to annotate text with ink. Since the layout is always the same, it&rsquo;s coherent to ink spatially on the rendered text. Ink wouldn&rsquo;t make sense on text that constantly changed layout.</p>
<figure><video alt="Text cards enforce a reasonable type size and line width independent of card dimensions or zoom." src="https://www.inkandswitch.com/muse/static/text-card-resize-and-zoom.mp4#t=0.01"></video>
<figcaption>
<h4>Text card resize and zoom</h4>
<p>Text cards enforce a reasonable type size and line width independent of card dimensions or zoom.</p>
</figcaption>
</figure>
<p>As the above video shows, text cards also don&rsquo;t follow the spatial model used in the rest of the app; they don&rsquo;t get bigger as you &ldquo;zoom&rdquo; in. This allowed us to read text on cards within boards. That&rsquo;s helpful when the text cards are used as labels and captions. But breaking the spatial model also reduces consistency.</p>
<p>We think more experimentation is needed on representing text in an app like Muse. The question of sizing / zooming needs more thought, and we&rsquo;d also like to consider rich text or Markdown, and how ink annotations would interact with editable text.</p>
<p>The <a href="https://www.inkandswitch.com/capstone-manuscript.html#stylus-mode-switching">biggest unsolved design problem from our previous prototype</a> was tool selection with the stylus. That is, how do you switch between functions like moving cards, inking, creating boards, and erasing?</p>
<p>There are two constraints that, if respected, make this stylus tool selection problem hard. The first constraint is not cluttering the user&rsquo;s screen with chrome. In Muse we wanted 100% of the screen to be devoted to the user&rsquo;s content at all times. So no tool palettes, buttons, menus, etc. (the obvious UIs for tool selection).</p>
<figure><img alt="It&rsquo;s easy to get lots of functionality in a stylus based app via chrome like buttons, palettes, and menus. Procreate, shown here, asks the users to select from dozens of potential brushes, obscuring large parts of their work." src="https://www.inkandswitch.com/muse/static/procreate-chrome.jpg">
<figcaption>
<h4>Tool chrome in Procreate</h4>
<p>It&rsquo;s easy to get lots of functionality in a stylus based app via chrome like buttons, palettes, and menus. <a href="https://procreate.art">Procreate</a>, shown here, asks the users to select from dozens of potential brushes, obscuring large parts of their work.</p>
</figcaption>
</figure>
<p>The second constraint is not creating a modal interface. By this we mean that the result of a gesture shouldn&rsquo;t depend on state that isn&rsquo;t the immediate focus of the user&rsquo;s attention. Thus buttons on the top of the screen that toggled between (say) inking and erasing would be modal, because when the user goes to ink or erase their locus of attention will no longer be on the buttons, but on the page they&rsquo;re about to draw on.</p>
<figure><img alt="In Goodnotes on iPad, whether the Apple Pencil inks or erases depends on the state of the buttons at the edge of the screen (top highlight), which tend to be far away from where the user is actually working (bottom highlight). Users often forget the state and ink when they meant to erase, or vice versa. Further reading on mode and mode errors in The Humane Interface, chapter 3. " src="https://www.inkandswitch.com/muse/static/goodnotes-modal.jpg">
<figcaption>
<h4>Mode errors</h4>
<p>In Goodnotes on iPad, whether the Apple Pencil inks or erases depends on the state of the buttons at the edge of the screen (top highlight), which tend to be far away from where the user is actually working (bottom highlight). Users often forget the state and ink when they meant to erase, or vice versa.</p>
<p>Further reading on mode and mode errors in <a href="https://www.academia.edu/2148659/The_humane_interface_new_directions_for_designing_interactive_systems">The Humane Interface</a>, chapter 3.</p>
</figcaption>
</figure>
<p>The solution we arrived at in Muse is to use physically distinct stylus grips&mdash;and resulting stylus altitudes&mdash;to provide more differentiation of gestures.</p>
<figure>
<figcaption>
<h4>Two grips</h4>
<p>The two types of stylus grips used in Muse.</p>
<p>A standard writing grip (left) gives a high altitude to the stylus. A low-angle grip (right) requires a reconfiguration of the hand and results in a low altitude.</p>
</figcaption>
</figure>
<p>The key to this design is that the two grips are a) physically distinct, with a discrete motion required to move from one two the other, and b) force different altitudes for the stylus. This means that users can habituate to the different gestures (essentially, train muscle memory) and that Muse can differentiate between them based on altitude. We combine this differentiation with the content that the user is focusing on in the operation to give the following full set of operations in Muse.</p>
<figure>
<figcaption>
<p>A combination of context and gesture determine the operation invoked.</p>
</figcaption>
<table>
<thead>
<tr>
<th>
<p>Context</p>
</th>
<th>
<p>Gesture</p>
</th>
<th>
<p>Operation</p>
</th>
</tr>
</thead>
<tbody>
<tr>
<td rowspan="2">
<p>Anywhere</p>
</td>
<td>
<p>Stylus writing grip</p>
</td>
<td>
<p>Ink</p>
</td>
</tr>
<tr>
<td>
<p>Stylus writing grip + finger touching screen</p>
</td>
<td>
<p>Erase</p>
</td>
</tr>
<tr>
<td>
<p>Edge of board</p>
</td>
<td rowspan="4">
<p>Low-angle grip</p>
</td>
<td>
<p>Create / delete card</p>
</td>
</tr>
<tr>
<td>
<p>Middle of card</p>
</td>
<td>
<p>Move card</p>
</td>
</tr>
<tr>
<td>
<p>Corner of card</p>
</td>
<td>
<p>Resize card</p>
</td>
</tr>
<tr>
<td>
<p>PDF</p>
</td>
<td>
<p>Create excerpt</p>
</td>
</tr>
<tr>
<td>
<p>Middle of card</p>
</td>
<td>
<p>Low-angle grip + finger touching screen</p>
</td>
<td>
<p>Copy card</p>
</td>
</tr>
</tbody>
</table>
</figure>
<p>After some time using the app, users develop habits around the different operations and move between them without conscious interruption.</p>
<figure><video alt="A user inks and then moves some cards. Both operations happen on a board, but require a physically distinct grip to which the user can habituate." src="https://www.inkandswitch.com/muse/static/grip-change.mp4#t=0.01"></video>
<figcaption>
<h4>Grip change</h4>
<p>A user inks and then moves some cards. Both operations happen on a board, but require a physically distinct grip to which the user can habituate.</p>
</figcaption>
</figure>
<p>Creating a modeless interface made a big difference in usability. In both Muse and our previous Capstone project, <a href="https://www.inkandswitch.com/capstone-manuscript.html#stylus-tool-selection">modal stylus interfaces inevitably left users frustrated</a>. We tried variations on modal interfaces in Muse including the one below, they but they were all just unsatisfactory to varying degrees.</p>
<figure><video alt="An earlier version of Muse used a hotspot interface to switch between inking and moving cards. This didn&rsquo;t require any chrome on the screen. But users forgot what mode they were in or got distracted from their work trying to remember it." src="https://www.inkandswitch.com/muse/static/mode-switch-hotspot.mp4#t=0.01"></video>
<figcaption>
<h4>Hotspot</h4>
<p>An earlier version of Muse used a hotspot interface to switch between inking and moving cards. This didn&rsquo;t require any chrome on the screen. But users forgot what mode they were in or got distracted from their work trying to remember it.</p>
</figcaption>
</figure>
<figure><video alt="Another earlier Muse experiment used two fingers touching the screen to indicate a different stylus operation. However, unlike the one finger down used for erasing, two fingers down isn&rsquo;t comfortable in many of the tablet grips commonly used with an iPad Pro. We think using multiple fingers of the non-stylus hand is an interesting future direction, but may require tradeoffs such as using the tablet with a more circumscribed set of grips." src="https://www.inkandswitch.com/muse/static/mode-switch-twofinger.mp4#t=0.01"></video>
<figcaption>
<h4>Two-finger tap</h4>
<p>Another earlier Muse experiment used two fingers touching the screen to indicate a different stylus operation. However, unlike the one finger down used for erasing, two fingers down isn&rsquo;t comfortable in many of the tablet grips commonly used with an iPad Pro. We think using multiple fingers of the non-stylus hand is an interesting future direction, but may require tradeoffs such as using the tablet with a more circumscribed set of grips.</p>
</figcaption>
</figure>
<p>We think the main takeaway from this experiment in Muse is that modeless stylus interfaces are preferable to modal ones, and that they are possible in practice. A secondary lesson is that distinct stylus grips are a plausible mechanism for achieving this in future projects.</p>
<p>We expect that more exploration of modeless stylus interfaces will yield further options (especially if more hardware options become available, such as a second stylus for the device). Specific future designs will also depend on the exact set of operations desired, as each new operation threatens to exhaust the available gesture space and thereby require a total change of the design.</p>
<p>There are a few important problems we either didn&rsquo;t get to address in Muse or for which we didn&rsquo;t find a satisfactory answer. We hope to address these problems in future work. If you have insights on these or would like to collaborate, let us know!</p>
<p>One thing that the iPad Pro doesn&rsquo;t solve is the <a href="https://www.inkandswitch.com/capstone-manuscript.html#problems-with-lost-styluses">problem of lost or broken styluses</a>. Several of our test users had issues here: for example, one user&rsquo;s stylus broke on a flight immediately before their user test. We think tools that require a stylus are still viable, but will be troublesome until platforms make it easier for users to always have a working stylus. The most recent iPad Pros do have a magnetic attachment for a stylus on the side of the device, which anecdotally helps with this problem.</p>
<figure><img alt="On the newest iPad Pros, the stylus attaches magnetically to the side of the tablet. This helps keep the tablet and stylus together without requiring additional hardware such as a stylus loop." src="https://www.inkandswitch.com/muse/static/stylus-attach.jpg">
<figcaption>
<h4>Stylus attachment</h4>
<p>On the newest iPad Pros, the stylus attaches magnetically to the side of the tablet. This helps keep the tablet and stylus together without requiring additional hardware such as a <a href="https://www.amazon.com/s?k=stylus+loop&amp;ref=nb_sb_noss_2">stylus loop</a>.</p>
</figcaption>
</figure>
<p>Right now it&rsquo;s too hard to get documents into Muse. There are some obvious things we could do here: implementation of mobile share sheets for ingestion, a more complete browser extension, a desktop import tool, etc. In the other direction, work is currently trapped in Muse and can&rsquo;t easily be used in a user&rsquo;s ongoing creative work. For example, if a user makes free-hand notes in Muse, it&rsquo;s hard to use those notes to compose an essay on a desktop. Again there are straightforward options here, like exporting from Muse to standard file formats.</p>
<p>That said, we think a future in which we create yet another data silo in an app like Muse is less interesting than one in which Muse natively integrates with users&rsquo; document storage systems of choice. Even with good import/export you are by definition constantly doing extra steps. Many of our target users use Dropbox or Google Drive to store their long-term creative work. So perhaps it&rsquo;s more interesting for an app like Muse to integrate with these data stores than to constantly import/export from its own data silo.</p>
<p>One area we didn&rsquo;t attempt to address was entering or editing text. This is a huge missing capability. Although a user can take handwritten notes, there are many circumstances where a few words or a sentence of structured plaintext would be more suitable. Consider also the need to make a few edits, even just formatting cleanup, to an existing text excerpt or text imported from some other source.</p>
<p>Touchscreen-only devices are poor for text entry and manipulation. Options to address this include an onscreen virtual keyboard, a wireless or folio physical keyboard, voice dictation, and handwriting recognition. All of these have major problems or weaknesses and we anticipate this is a major problem that must be solved to make a useful studio for ideas on a tablet.</p>
<p>We&rsquo;ll address this in future research. Consider this also an invitation to other researchers to tackle the problem; if you&rsquo;ve done so, we&rsquo;d love to hear from you.</p>
<p>A full-powered creative suite will work across multiple devices. Different form factors are better for different types of work, and the best tool sets will leverage that. For example, Muse shows how tablets are great for reading, annotating, and free-form ideating. However, phones are better for on-the-go lookups, and desktops are still best for complex research, authoring, and editing workflows.</p>
<p>With Muse, we mostly had to deal with one device (the web clipper is sort of a second device, but was minimal). To extend to multiple devices, there&rsquo;s an obvious technical challenge of synchronizing state. Our lab has worked on this problem for some time (e.g. with <a href="https://github.com/automerge/automerge">Automerge</a>), but it is far from solved. Another challenging problem is the UI: the spatial interface worked well on a tablet, but what would it mean for other devices? A phone is probably too small to legibly render a whole board. A desktop could render a board, but power users want to use keyboard shortcuts there, which don&rsquo;t obviously translate to spatial UIs.</p>
<figure>
<figcaption>
<p>A use case illustrating the strengths of different devices. The phone (top left) is used to capture a PDF found on the go for later reading. The tablet (top right) is used to read and annotate the PDF. The desktop (bottom) is used to review the annotated paper and use it while creating a presentation referencing its contents.</p>
</figcaption>
</figure>
<p>Is the answer to this challenge to implement a different metaphor for each form factor, according to the strengths of the device? For example, touch-based spatial navigation on tablets, simple lists on phones, and keyboard-navigable pages on desktops? This might optimize the interaction for each device, but also leave the user with an inconsistent mental model across their devices, inhibiting fluid thinking. But it would also be disappointing if the answer was a least-common-denominator approach across the different form factors.</p>
<p>In Muse it&rsquo;s possible to annotate / ink on any content type, including boards, images, PDFs, web pages, and text documents. For something like PDFs, which visually render in the same way everywhere, inking annotations make total sense. For example, if a user draws a red line in the upper-right corner of page 3 of a PDF on their tablet, it&rsquo;s clear how to render that annotation on another tablet, regardless of it&rsquo;s screen size, orientation, etc.</p>
<figure>
<figcaption>
<p>In the left image, the user annotated a web page with red ink. As long as the size of the rendered page and therefore the layout stays the same, it&rsquo;s clear how to render these annotations.</p>
<p>But what if the page is resized, as in the second image? The dashed lines indicate where one might expect the annotations to go. But it&rsquo;s very difficult for software to determine such a transformation for arbitrary free-form annotations."</p>
</figcaption>
</figure>
<p>It&rsquo;s less obvious what to do in general about e.g. web pages, because how these documents render depends on the user&rsquo;s device: their screen size and resolution, current font size, etc. So if a user annotates a web page with a line on one device, how should it be rendered elsewhere? There isn&rsquo;t an obvious answer. In Muse we&rsquo;ve avoided this problem by assuming the web page will only ever be rendered on one device (the user&rsquo;s tablet) and thus that it will be rendered the same every time.</p>
<p>A similar problem exists for text. Text can be rendered in different ways on different devices. Users also expect text to change (e.g. if you edit it in a text editor), providing another source of disorientation for the original annotations.</p>
<p>We think being able to annotate any document type is useful, but it will require careful design to be deployed in more general systems. Perhaps the answer is simply to freeze or copy any document before it is annotated. There may be better answers to this design problem, though.</p>
<p>We described earlier solutions for interpreting touch and stylus gestures without resorting to modes or interface chrome. But these solutions are sensitive to the total number of desired operations. If we wanted to add even one more operation to the tool, we may need to totally redesign the interface / gesture system. Even something as simple as a &ldquo;settings menu&rdquo; doesn&rsquo;t have an obvious place in our design right now.</p>
<p>We expect production-grade tools will require more operations than Muse currently supports. One way to support them would be to continue to design bespoke interfaces as operations are added, but there may be more scalable approaches. For example, in keyboard interfaces, the pattern of some set of modifier keys being used with single letter keys provides potential shortcuts for tens or even hundreds of operations. It may be desirable to find such an interface for touch devices. Options to explore include <a href="https://en.wikipedia.org/wiki/Chording#Multitouch_chording">chording</a> and auxiliary mechanical modifier keys.</p>
<p>A freeform digital space for idea development could be a boon to creative people. With the Muse studio for ideas, we sought to move away from the common qualities of digital workspaces (structured, neat, and sterile) and to capture some of the qualities of physical crafting spaces (messy, informal, and personal).</p>
<p>We think this project demonstrates promise along these lines, but many questions remain open. What do you think: did we achieve our goals, or do you think this goal is even worthwhile in the first place? Or perhaps you are working on a related project with your own insights to share. Either way we&rsquo;d love to hear from you: <a title="Tweet us @inkandswitch" href="https://twitter.com/inkandswitch">@inkandswitch</a> or <span><a title="Send us an Email" href="https://www.inkandswitch.com/cdn-cgi/l/email-protection#aec6cbc2c2c1eec7c0c5cfc0caddd9c7dacdc680cdc1c3"><span>[email&nbsp;protected]</span></a>.</span></p>
<p><em>Update August 2019: Based on the positive response to this article, we&rsquo;re working on Muse as a full product. If you&rsquo;d like to be a beta tester, <a href="https://museapp.com/">sign up here</a>.</em></p>
</article>
</div>
</div>
</div>
</div>
<br>
<br>
<br>
<br>
<br>
<br>
via Ink &amp; Switch<br>
<br>
December 2, 2022 at 04:40PM
